src/client/features/view/components/menu/search.js

```diff
@@ -0,0 +1,94 @@
+const _ = require('lodash');
+
+//Apply hover styling to a collection of nodes
+function updateStyling(style, matchedNodes) {
+  matchedNodes.forEach(node => {
+    let isCompartment = node.data('class') === 'compartment';
+
+    //Collapse Expanded Targets
+    if (node.isExpanded() && !isCompartment) { node.collapse(); }
+
+    //Apply style to the parent
+    if (node.isChild() && node.parent().data('class') !== 'compartment') { node = node.parent(); }
+
+    applySearchStyle(node, style);
+  });
+}
+
+//Determine if a regex pattern is valid
+function validateRegex(pattern) {
+  var parts = pattern.split('/'),
+    regex = pattern,
+    options = "";
+  if (parts.length > 1) {
+    regex = parts[1];
+    options = parts[2];
+  }
+  try {
+    let regexObj = new RegExp(regex, options);
+    return regexObj;
+  }
+  catch (e) {
+    return null;
+  }
+}
+
+function applySearchStyle(eles, style) {
+  eles.style(style);
+}
+
+//Search for nodes that match an entered query
+function searchNodes(query, cy) {
+  const searchValue = query.target.value;
```

Comment:
See above.

Reply:
Removed, as query is already the value in the newer commit